api/src/Entity/ContentNode.php

```diff
@@ -56,29 +53,17 @@ abstract class ContentNode extends BaseEntity implements BelongsToCampInterface,
     #[ApiProperty(writable: false, example: '/content_nodes/1a2b3c4d')]
     #[Gedmo\SortableGroup] // this is needed to avoid that all root nodes are in the same sort group (parent:null, slot: '')
     #[Groups(['read'])]
-    #[ORM\ManyToOne(targetEntity: ContentNode::class, inversedBy: 'rootDescendants')]
+    #[ORM\ManyToOne(targetEntity: ColumnLayout::class, inversedBy: 'rootDescendants')]
     #[ORM\JoinColumn(nullable: true)] // TODO make not null in the DB using a migration, and get fixtures to run
-    public ContentNode $root;
-
-    /**
-     * All content nodes that are part of this content node tree.
-     */
-    #[ApiProperty(readable: false, writable: false)]
-    #[ORM\OneToMany(targetEntity: ContentNode::class, mappedBy: 'root')]
-    public Collection $rootDescendants;
+    public ?ColumnLayout $root = null;
 
     /**
      * The parent to which this content node belongs. Is null in case this content node is the
      * root of a content node tree. For non-root content nodes, the parent can be changed, as long
-     * as the new parent is in the same camp as the old one. A content node is defined as root when
-     * it has an owner.
+     * as the new parent is in the same camp as the old one.
      */
-    #[AssertEitherIsNull(
-        other: 'owner',
-        messageBothNull: 'Must not be null on non-root content nodes.',
-        messageNoneNull: 'Must be null on root content nodes.'
-    )]
-    #[AssertBelongsToSameOwner(groups: ['update'])]
+    #[Assert\NotNull(groups: ['create'])] // Root nodes have parent:null, but manually creating root nodes is not allowed
```

Comment:
I think `#[AssertBelongsToSameRoot(groups: ['update'])]` already disallows changing parent to null (because that would implicitly change the root). There's also a test for this (testPatchColumnLayoutValidatesMissingParent).

But I think it doesn't hurt to `#[Assert\NotNull]`, that would make it more explicit.

Reply:
Ah, now I remember why this was not possible. If `#[Assert\NotNull]` is enabled for update, I cannot modify root-nodes anymore. So I decided to to verify this in `#[AssertBelongsToSameRoot]`. Basically, changing the `parent` property from non-null to null would make the `ContentNode` a root and therefore implictly also changing the root (which is not allowed according to `#[AssertBelongsToSameRoot]`).

Happy to discuss counterproposals.

Reply:
> I think you need to revert this to use `AssertEitherIsNull` again.

There's no `owner` anymore. Before there were 2 evidences that a `ContentNode` is a root (`parent` is null and `owner` is not-null). They had to be in sync with `AssertEitherIsNull`. Now there's only parent and nothing to verify/sync against.